### PR TITLE
Add implementation for env.shell plugin API

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

#### What it does

This PR provides an implementation for [env.shell](https://code.visualstudio.com/api/references/vscode-api#env) VS Code API. 

It is needed for https://github.com/eclipse/che/issues/15116

#### How to test

It can be tested with [VS Code Quarkus](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-quarkus) extension. 

1. Download the Quarkus extension and put it into /plugins folder
2. Open some java project, for example https://github.com/quarkusio/quarkus-quickstarts/tree/master/getting-started
3. Go to **Quarkus: Add extension to currant project** command
4. Select some extensions and press Enter

The terminal will be opened and extension should start installing
#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/plugin/terminal-ext.ts [hunk 1]
```diff
@@ -115,6 +115,15 @@ export class TerminalServiceExtImpl implements TerminalServiceExt {
         this.onDidChangeActiveTerminalEmitter.fire(this.activeTerminal);
     }
 
+    private defauleShell: string;
```
Comment (benoitf) [inline packages/plugin-ext/src/plugin/terminal-ext.ts]:
```suggestion
    private defaultShell: string;
```

File: packages/plugin-ext/src/main/browser/terminal-main.ts [hunk 2]
```diff
@@ -44,6 +44,9 @@ export class TerminalServiceMainImpl implements TerminalServiceMain, Disposable
         }
         this.toDispose.push(this.terminals.onDidChangeCurrentTerminal(() => this.updateCurrentTerminal()));
         this.updateCurrentTerminal();
+        this.terminals.getDefaultShell().then(shell => {
```
Comment (akosyakov) [inline packages/plugin-ext/src/main/browser/terminal-main.ts]: Could not it be a race? If an extension reads `shell` before this method resolves?

Probably the better place to pass it here during initialization: https://github.com/svor/theia/blob/0e002ce2d8baa8fe9a90245912fa20e412d48e2c/packages/plugin-ext/src/hosted/browser/hosted-plugin.ts#L403

Comment (svor) [reply packages/plugin-ext/src/main/browser/terminal-main.ts]: makes sense, moved the initialization into hosted-plugin.

File: packages/plugin-ext/src/hosted/browser/hosted-plugin.ts [hunk 3]
```diff
@@ -404,7 +408,11 @@ export class HostedPluginSupport {
                 preferences: getPreferences(this.preferenceProviderProvider, this.workspaceService.tryGetRoots()),
                 globalState,
                 workspaceState,
-                env: { queryParams: getQueryParameters(), language: navigator.language },
+                env: {
+                    queryParams: getQueryParameters(),
+                    language: navigator.language,
+                    shell: await this.terminalService.getDefaultShell()
```
Comment (akosyakov) [inline packages/plugin-ext/src/hosted/browser/hosted-plugin.ts]: could be mode to `Promise.all` above to fetch it in parallel

Comment (svor) [reply packages/plugin-ext/src/hosted/browser/hosted-plugin.ts]: Moved. Thank you for reviewing!

**Comment by akosyakov** [comment]
please merge as soon as build is green, recently it takes a while

**Pull request merged**